torchmetrics/image/kid.py

```diff
@@ -0,0 +1,284 @@
+            valid_int_input = ['logits_unbiased', 64, 192, 768, 2048]
+            if feature not in valid_int_input:
+                raise ValueError(
+                    f'Integer input to argument `feature` must be one of {valid_int_input},'
+                    f' but got {feature}.'
+                )
+
+            self.inception = NoTrainInceptionV3(name='inception-v3-compat', features_list=[str(feature)])
+        elif isinstance(feature, torch.nn.Module):
+            self.inception = feature
+        else:
+            raise ValueError('Got unknown input to argument `feature`')
+
+        if not (isinstance(subsets, int) and subsets > 0):
+            raise ValueError("Argument `subsets` expected to be integer larger than 0")
+        self.subsets = subsets
+
+        if not (isinstance(subset_size, int) and subset_size > 0):
+            raise ValueError("Argument `subset_size` expected to be integer larger than 0")
+        self.subset_size = subset_size
+
+        if not (isinstance(degree, int) and degree > 0):
+            raise ValueError("Argument `degree` expected to be integer larger than 0")
+        self.degree = degree
+
+        if gamma is not None and not isinstance(gamma, float):
+            raise ValueError("Argument `gamma` expected to be `None` or float larger than 0")
+        self.gamma = gamma
+
+        if not (isinstance(coef, float) and coef > 0):
+            raise ValueError("Argument `coef` expected to be float larger than 0")
+        self.coef = coef
+
+        # states for extracted features
+        self.add_state("real_features", [], dist_reduce_fx=None)
+        self.add_state("fake_features", [], dist_reduce_fx=None)
+
+    def update(self, imgs: Tensor, real: bool) -> None:  # type: ignore
+        """ Update the state with extracted features
+
+        Args:
+            imgs: tensor with images feed to the feature extractor
+            real: bool indicating if imgs belong to the real or the fake distribution
+        """
+        features = self.inception(imgs)
+
+        if real:
+            self.real_features.append(features)
+        else:
+            self.fake_features.append(features)
+
+    def compute(self) -> Tuple[Tensor, Tensor]:
+        """ Calculate KID score based on accumulated extracted features from the two distributions.
+            Returns a tuple of mean and standard deviation of KID scores calculated on subsets of
+            extracted features.
+
+            Implementation inspired by https://github.com/toshas/torch-fidelity/blob/v0.3.0/torch_fidelity/metric_kid.py
+        """
+        real_features = torch.cat(self.real_features, dim=0)
+        fake_features = torch.cat(self.fake_features, dim=0)
+
+        n_samples_real = real_features.shape[0]
+        if n_samples_real < self.subset_size:
+            raise ValueError()
+        n_samples_fake = fake_features.shape[0]
+        if n_samples_fake < self.subset_size:
+            raise ValueError()
```

Comment:
Meaningful description is missing here as well :)

Reply:
Done :]